src/components/pages/Listings/Confirm.tsx

```diff
@@ -0,0 +1,263 @@
+            {type !== 'english' && (
+              <Button
+                variant="secondary"
+                className="w-full"
+                onClick={() =>
+                  navigate(listingId ? `/listings/${listingId}` : '/listings')
+                }
+              >
+                View this listing
+              </Button>
+            )}
+            <Button
+              variant="secondary"
+              className="w-full"
+              onClick={() => navigate(`/listings`)}
+            >
+              Go to marketplace
+            </Button>
+          </div>
+        </div>
+      </>
+    );
+  }
+
+  return (
+    <>
+      <GoBackHeader
+        title="Confirm purchase"
+        className="w-full my-12"
+        onGoBack={() => {
+          navigate(listingId ? `/listings/${listingId}` : '/listings');
+        }}
+      />
+      <div className="max-w-2xl w-full px-6 mx-auto pb-12">
+        <Card className="flex flex-col gap-6">
+          <Row label="Domain name" value={name} variant="large" />
+          <Row label="Price" value={`${price} ARIO`} />
+          <div className="flex gap-2 justify-end mt-4">
+            <Button
+              variant="outline"
+              size="small"
+              onClick={() => {
+                navigate(listingId ? `/listings/${listingId}` : '/listings');
+              }}
+            >
+              Cancel
+            </Button>
+            <Button
+              variant="primary"
+              size="small"
+              disabled={
+                !walletAddress || isMutationPending || isMutationSuccess
+              }
+              onClick={() => {
+                const operation = type === 'english' ? 'bid' : 'buy';
+                const mutation =
+                  operation === 'bid' ? mutationBidListing : mutationBuyListing;
+
+                if (!price) {
+                  eventEmitter.emit('error', {
+                    name: 'Validation',
+                    message: 'Price is required',
+                  });
+
+                  return;
+                }
+
+                mutation.mutate(
+                  { price },
+                  {
+                    onError: (error) => {
+                      eventEmitter.emit('error', {
+                        name: `Failed to ${operation}`,
+                        message: error.message,
+                      });
+                    },
+                    onSuccess: async (data) => {
+                      console.log(`${operation} success`, { data });
+                      void Promise.allSettled([
+                        queryClient.refetchQueries({
+                          queryKey: [marketplaceQueryKeys.listings.all],
+                        }),
+                        queryClient.refetchQueries({
+                          queryKey: [marketplaceQueryKeys.myANTs.all],
+                        }),
+                      ]);
+                    },
```

Comment:
see comment regarding reusable mutations - this can be defined on the use[my-mutation] hook and then you just call mutate(params) i believe